app/models/government.rb

```diff
@@ -20,4 +22,51 @@ def self.on_date(date)
   def current?
     self == Government.current
   end
+
+  def overlaps?(other)
+    !(is_before?(other) || is_after?(other))
+  end
+
+private
+
+  def is_before?(other)
+    starts_before?(other) && ends_before?(other)
+  end
+
+  def is_after?(other)
+    starts_after?(other) && ends_after?(other)
+  end
+
+  def starts_before?(other)
+    start_date < other.start_date
+  end
+
+  def ends_before?(other)
+    (end_date || distant_future) < other.start_date
```

Comment:
Maybe we should also extract an `has_ended?` method: `has_ended? && end_date < other.start_date`.